ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/example/OpenAPIExampleMapper.java

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.openapi.service.mapper.example;
+
+/**
+ * This {@link OpenAPIExampleMapper} class represents the interface for mapping all the examples in the OpenAPI
+ * specification.
+ *
+ * @since 2.1.0
+ */
+public interface OpenAPIExampleMapper {
```

Comment:
we have `ExampleMapper` and `OpenAPIExampleMapper` interfaces and both have `setExamples()`. What is the difference between these?

Reply:
The `ExampleMapper` is an abstract class which represents the class which maps `@openapi:Example` annotation into the OpenAPI specification. The `TypeExampleMapper` is one of its implementation. It only has a `setExample` method which adds the example value to the schema.

There is another abstract class called `ExamplesMapper`(extends the `ExampleMapper`) which represents the class which maps `@openapi:Example` and `@openapi:Examples` annotations into the OpenAPI specification. It has `setExamples` method which adds the example values to the schema and also has `setExample` from the `ExampleMapper`.

The `OpenAPIExampleMapper` is a interface which takes the entire OpenAPI model and maps the examples for components, parameters and request body by instantiating the necessary  `ExampleMapper` or `ExamplesMapper`.